internal/provider/gateway_fqdn.go

```
@@ -35,16 +36,22 @@ func NewGatewayFQDNDeployer(ctx context.Context, d *schema.ResourceData, apiClie
 	for idx, n := range backendsIf {
 		backends[idx] = zos.Backend(n.(string))
 	}
-	nodeDeploymentIDIf := d.Get("node_deployment_id").(map[string]interface{})
-	nodeDeploymentID := make(map[uint32]uint64)
-	for node, id := range nodeDeploymentIDIf {
-		nodeInt, err := strconv.ParseUint(node, 10, 32)
+	capacityID := d.Get("capacity_id").(uint64)
+	contractDeploymentMapIf := d.Get("capacity_deployment_map").(map[string]interface{})
+	capacityDeploymentMap := make(map[uint64]uint64)
+	for contractID, deploymentID := range contractDeploymentMapIf {
+		contractIDInt, err := strconv.ParseUint(contractID, 10, 64)
 		if err != nil {
-			return GatewayFQDNDeployer{}, errors.Wrap(err, "couldn't parse node id")
+			return GatewayFQDNDeployer{}, errors.Wrapf(err, "couldn't parse capacity reservation contract id %s", contractID)
 		}
-		deploymentID := uint64(id.(int))
-		nodeDeploymentID[uint32(nodeInt)] = deploymentID
+		deploymentIDInt := uint64(deploymentID.(int))
+		capacityDeploymentMap[contractIDInt] = deploymentIDInt
 	}
+	contract, err := apiClient.substrateConn.GetContract(capacityID)
+	if err != nil {
+		return GatewayFQDNDeployer{}, errors.Wrapf(err, "couldn't get capacity reservation contract %d info", capacityID)
+	}
+	nodeID := uint32(contract.ContractType.CapacityReservationContract.NodeID)
 	ncPool := client.NewNodeClientPool(apiClient.rmb)
 	deploymentData := DeploymentData{
 		Name:        d.Get("name").(string),
```

should it return an error instead of logging it in line 63?